src/main/java/mahesh/util/Ui.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    /**
+     * Prints a message indicating that a task has been deleted.
+     *
+     * @param task the task that was deleted
+     * @param taskCount the current number of tasks in the list
+     */
+    public static void printTaskDeleted(Task task, int taskCount) {
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("  " + task);
+        System.out.println("Now you have " + taskCount + " tasks in the list.");
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints a message indicating that a task has been marked as done.
+     *
+     * @param task the task that was marked as done
+     */
+    public static void printMarkedAsDone(Task task) {
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("  " + task);
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints a message indicating that a task has been unmarked as done.
+     *
+     * @param task the task that was unmarked as done
+     */
+    public static void printUnmarkedAsDone(Task task) {
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println("  " + task);
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints an error message.
+     *
+     * @param err the exception containing the error message
+     */
+    public static void printErr(MaheshException err) {
+        System.out.println(err.getMessage());
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints an error message indicating that there is no such task.
+     *
+     * @param taskCount the current number of tasks in the list
+     */
+    public static void printNoSuchTaskErr(int taskCount) {
+        System.out.println("There is no such task. You currently have " + taskCount + " task(s).");
+        System.out.println("Use the \"list\" command to view all your tasks.");
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints an error message indicating that the command is incomplete or incorrect.
+     *
+     * @param err the exception containing the error message
+     */
+    public static void printIncompleteCommandErr(MaheshException err) {
+        System.out.println("The command is incomplete/incorrect.");
+        System.out.println(err.getMessage());
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints an error message indicating that the task list is empty.
+     */
+    public static void printEmptyListErr() {
+        System.out.println("You have no tasks! Add a few tasks (todo, deadline or event)");
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Prints an error message indicating that some tasks may not have loaded correctly due to a corrupted data file.
+     */
+    public static void printCorruptedDataErr() {
+        System.out.println("Some tasks may not have loaded correctly due to corrupted data file.");
+        System.out.println(DIVIDER);
```

Comment:
Since you've defined `printDivider()`, maybe you could consider using in your other methods. This could be useful if you ever chose to add another print statement to your divider!